test/contlinear.jl

```diff
@@ -726,6 +754,241 @@ function linear7test(solver::MOI.AbstractSolver, ε=Base.rtoldefault(Float64))
+        if MOI.getattribute(m, MOI.ResultCount()) == 1
+            # solver returned an unbounded ray
+            @test MOI.getattribute(m, MOI.TerminationStatus()) == MOI.Success
+            @test MOI.getattribute(m, MOI.PrimalStatus()) == MOI.InfeasibilityCertificate
+            @test MOI.cangetattribute(m, MOI.VariablePrimal(), [x, y])
+            ray = MOI.getattribute(m, MOI.VariablePrimal(), [x,y])
+            @test ray[1] ≈ ray[2] atol=ε
+
+        else
+            # solver returned nothing
+            @test MOI.getattribute(m, MOI.ResultCount()) == 0
+            @test MOI.cangetattribute(m, MOI.PrimalStatus(1)) == false
+            @test MOI.getattribute(m, MOI.TerminationStatus()) == MOI.UnboundedNoResult ||
+                MOI.getattribute(m, MOI.TerminationStatus()) == MOI.InfeasibleOrUnbounded
+        end
+    end
+end
+
+function linear9test(solver::MOI.AbstractSolver, ε=Base.rtoldefault(Float64))
+    @testset "addconstraints" begin
+        #   maximize 1000 x + 350 y
+        #
+        #       s.t.                x >= 30
+        #                           y >= 0
+        #                 x -   1.5 y >= 0
+        #            12   x +   8   y <= 1000
+        #            1000 x + 300   y <= 70000
+        #
+        #   solution: (59.0909, 36.3636)
+        #   objv: 71818.1818
+        @test MOI.supportsproblem(solver, MOI.ScalarAffineFunction{Float64},
+            [
+                (MOI.ScalarAffineFunction{Float64},MOI.GreaterThan{Float64}),
+                (MOI.ScalarAffineFunction{Float64},MOI.LessThan{Float64}),
+                (MOI.SingleVariable,MOI.GreaterThan{Float64})
+            ]
+        )
+
+        m = MOI.SolverInstance(solver)
+        x = MOI.addvariable!(m)
+        y = MOI.addvariable!(m)
+
+        MOI.addconstraints!(m,
+            [MOI.SingleVariable(x), MOI.SingleVariable(y)],
+            [MOI.GreaterThan(30.0), MOI.GreaterThan(0.0)]
+        )
+
+        MOI.addconstraints!(m,
+            [MOI.ScalarAffineFunction([x, y], [1.0, -1.5], 0.0)],
+            [MOI.GreaterThan(0.0)]
+        )
+
+        MOI.addconstraints!(m,
+            [
+                MOI.ScalarAffineFunction([x, y], [12.0, 8.0], 0.0),
+                MOI.ScalarAffineFunction([x, y], [1_000.0, 300.0], 0.0)
+            ],
+            [
+                MOI.LessThan(1_000.0),
+                MOI.LessThan(70_000.0)
+            ]
+        )
+
+        MOI.setobjective!(m, MOI.MaxSense,
+            MOI.ScalarAffineFunction([x, y], [1_000.0, 350.0], 0.0)
+        )
+
+        MOI.optimize!(m)
+
+        @test MOI.getattribute(m, MOI.TerminationStatus()) == MOI.Success
+        @test MOI.getattribute(m, MOI.PrimalStatus()) == MOI.FeasiblePoint
+
+        @test MOI.getattribute(m, MOI.ObjectiveValue()) ≈ 79e4/11 atol=ε
+        @test MOI.getattribute(m, MOI.VariablePrimal(), x) ≈ 650/11 atol=ε
+        @test MOI.getattribute(m, MOI.VariablePrimal(), y) ≈ 400/11 atol=ε
```

Comment:
I would use `rtol` instead of `atol` here. `isapprox` is implemented as
```julia
abs(x-y) <= atol + rtol*max(abs(x), abs(y)))
```
where `rtol` is `1.4901161193847656e-8` for `Float64` by default.
For the objective `rtol * max(abs(x), abs(y))` is approximately 1e-3 so if `atol` is like `1e-5` it will be neglibeable.
CSDP fails the test for `VariablePrimal`/`x` because epsilon is 1e-7 which means that `atol` is 1e-7` but `rtol * max(abs(x), abs(y))` is about `1e-7` too. CSDP has a relative tolerance of `1e-7` so it should pass the test but here it is tested for an *absolute* tolerance of 1e-7.

In fact, it makes more sense to only use `atol` when one of the two numbers being compared is 0 or very close to 0. Why we use `atol` everywhere in MOI's tests ? Usually it does not matter much but here since the numbers are large, it does matter.